docs/guide/search/extending_search/custom_search.md

```diff
@@ -0,0 +1,79 @@
+# Implement custom search
+
+This example shows how to create a new service which searches for products that:
```

Comment:
```suggestion
This example shows how to create a new service which searches for products that are:
```
Does it search for products that are discontinued and visible at the same time? Or discontinued OR visible separately? Or any combination?